src/main/java/com/nutomic/syncthingandroid/syncthing/NetworkReceiver.java

```diff
@@ -13,19 +14,27 @@
 public class NetworkReceiver extends BroadcastReceiver {
 
     private static final String TAG = "NetworkReceiver";
+    private static final int WIFI_AP_STATE_ENABLED = 13;
 
     @Override
     public void onReceive(Context context, Intent intent) {
         if (!SyncthingService.alwaysRunInBackground(context))
             return;
 
-        ConnectivityManager cm =
-                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-        NetworkInfo wifiInfo = cm.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
-        boolean isWifiConnected = (wifiInfo != null && wifiInfo.isConnected());
-        Log.v(TAG, "Received wifi " + (isWifiConnected ? "connected" : "disconnected") + " event");
         Intent i = new Intent(context, SyncthingService.class);
-        i.putExtra(DeviceStateHolder.EXTRA_HAS_WIFI, isWifiConnected);
+        String action = intent.getAction();
+        if (action.equals("android.net.wifi.WIFI_AP_STATE_CHANGED")) {
+            int state = intent.getIntExtra("wifi_state", 0);
```

Comment:
Something that is not enabled... disabled is 11 see https://github.com/android/platform_frameworks_base/blob/master/wifi/java/android/net/wifi/WifiManager.java#L206